### PR TITLE
[ssh] Collect sshd_config.d directory

### PR DESCRIPTION
**Pull request by pmoravec**
Collect sshd config files in its dedicated sshd_config.d directory, as well as config files included from them.

Resolves: #3595

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [X] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

**Comment by pmoravec** [comment]
The implementation (of traversing the directory within `included_configs`) is not ideal: it skips hidden files (until we support Python 3.11+ that [supports](https://docs.python.org/3/library/glob.html#glob.glob) `include_hidden` parameter). I failed to find any more elegant solution covering also that, though.

**Review by arif-ali — APPROVED** [review]

File: sos/report/plugins/ssh.py [hunk 1]
```diff
@@ -33,7 +34,8 @@ def setup(self):
 
         sshcfgs = [
             "/etc/ssh/ssh_config",
-            "/etc/ssh/sshd_config"
+            "/etc/ssh/sshd_config",
+            "/etc/ssh/sshd_config.d/*",
             ]
```
Comment (jcastill) [inline sos/report/plugins/ssh.py:39]: I ran an "old" sos (4.6) and it seems that we capture the .d directories already:
```
# ls /var/tmp/sosreport-localhost-2024-04-15-rxpnjai/etc/ssh/
ssh_config  ssh_config.d  sshd_config  sshd_config.d
```

And in the logs:

```
2024-04-15 09:58:56,424 INFO: [plugin:ssh] collecting path '/etc/ssh/sshd_config.d/50-redhat.conf'
2024-04-15 09:58:56,424 DEBUG: [plugin:ssh] copying path '/etc/ssh/sshd_config.d/50-redhat.conf' to archive:'/etc/ssh/sshd_config.d/50-redhat.conf'
2024-04-15 09:58:56,425 DEBUG: [archive:TarFileArchive] Making leading paths for /etc/ssh/sshd_config.d
```

Could be that when we run through sshcfgs below we take these as the beginning of the strings to check?

Comment (pmoravec) [reply sos/report/plugins/ssh.py:39]: Interesting.. Well, the `50-redhat.conf` is collected due to:

    Include /etc/ssh/sshd_config.d/*.conf

in `/etc/ssh/sshd_config`. Anyway I tested the file is honoured by `sshd` even without the `Include` line (and e.g. `https://manpages.debian.org/unstable/openssh-server/sshd_config.5.en.html` does suggest so as well).

Due to that "default include", it makes sense to dollect the dir (or just `*.conf` files..?), I think.

**Review by TurboTurtle — APPROVED** [review]

File: sos/report/plugins/ssh.py [hunk 2]
```diff
@@ -49,7 +51,12 @@ def included_configs(self, sshcfgs):
         """ Include subconfig files """
         # Read configs for any includes and copy those
         try:
-            for sshcfg in sshcfgs:
+            cfgfiles = [
+                f for files in [
+                    glob(copyspec, recursive=True) for copyspec in sshcfgs
+                ] for f in files
+            ]
```
Comment (TurboTurtle) [inline sos/report/plugins/ssh.py:58]: I think this could be cleaned up slightly, but it does track as it is.

LGTM either way.

**Pull request merged**